File: backend/agent-builder-service/app/core/permissions.py
"""Permission code constants this service checks against the `permissions` claim of a
verified iam-service access token. These strings must match iam-service's seeded catalog
exactly (see iam-service/scripts/seed_permissions_and_roles.py) - they are not derived or
guessed, they are looked up 1:1 against that source of truth.
"""

MODELS_MANAGE = "talentos.agentbuilder.models.manage"
AGENTS_READ = "talentos.agentbuilder.agents.read"
AGENTS_WRITE = "talentos.agentbuilder.agents.write"
AGENTS_PUBLISH = "talentos.agentbuilder.agents.publish"
AGENTS_MANAGE_KEYS = "talentos.agentbuilder.agents.manage_keys"

# This service's own machine identity uses this permission to manage service principals in
# iam-service on behalf of publish/regenerate - never checked against an end user's token.
IAM_SERVICE_PRINCIPALS_MANAGE = "talentos.iam.service_principals.manage"

File: frontend/agent-builder-console/src/lib/permissions.ts
/**
 * Permission codes this console gates its UI on - taken verbatim from agent-builder-service's
 * API contract (every endpoint except `GET /health` requires one of these; see that service's
 * README "API" table for the exact per-route mapping).
 */
export const PERMISSIONS = {
  MODELS_MANAGE: "talentos.agentbuilder.models.manage",
  AGENTS_READ: "talentos.agentbuilder.agents.read",
  AGENTS_WRITE: "talentos.agentbuilder.agents.write",
  AGENTS_PUBLISH: "talentos.agentbuilder.agents.publish",
  AGENTS_MANAGE_KEYS: "talentos.agentbuilder.agents.manage_keys",
} as const;

export { hasAnyPermission, hasPermission } from "./auth";
